src/riscv.c

```diff
@@ -293,14 +293,26 @@ static void load_dtb(char **ram_loc, vm_attr_t *attr)
 {
 #include "minimal_dtb.h"
     char *bootargs = attr->data.system.bootargs;
-    char *vblk = attr->data.system.vblk_device;
+    char **vblk = attr->data.system.vblk_device;
     char *blob = *ram_loc;
     char *buf;
     size_t len;
     int node, err;
     int totalsize;
 
-    memcpy(blob, minimal, sizeof(minimal));
+#define DTB_EXPAND_SIZE 1024 /* or more if needed */
+
+    /* Allocate enough memory for DTB + extra room */
+    int minimal_len = ARRAY_SIZE(minimal);
```

Comment:
Nit: Since this variable represents a length, would size_t be more appropriate than int?